content-src/components/Sections/Sections.jsx

```diff
@@ -306,14 +306,32 @@ export class _Sections extends React.PureComponent {
     return sections;
   }
 
+  renderComponent(component) {
+    const {props} = component;
+    if (component.type === "TopicsHeader") {
```

Comment:
I imagine we would expand the if here in some way to handle more components, dunno if we want to key the component names, seems logical, but for now I'm not making that assumption.

Reply:
Having to have a separate dispatcher just to manage the prop-passing seems a bit surprising, I wonder if there's a way to make this simpler.  Maybe it's something we can get redux to do for us?  That said, an XXX comment about this might be the right thing for now.

Reply:
Hm, what do you mean by dispatcher here?

Reply:
I'm assuming dmose was referring to the duplication of reading out each property to pass it on. You could probably do something like:
```js
const validComponents = {
  TopicsHeader
};
const extraProperties = {};
switch (type) {
  case "TopicsHeader":
    extraProperties.topics = …
    break;
}
return (<validComponents[type] {...properties} {...extraProperties} />);
```